src/Stack/Config.hs

```diff
@@ -275,7 +275,7 @@ loadBuildConfig menv mproject config stackRoot mresolver noConfigStrat = do
       Nothing -> case noConfigStrat of
         ThrowException -> do
             currDir <- getWorkingDir
-            cabalFiles <- findCabalFiles currDir
+            cabalFiles <- findCabalFiles True currDir
```

Comment:
I just passed in True here, can someone have a look if this might be problematic if `init` was called with the `--ignore-subdirs` flag?